src/openMVG/sfm/sfm_data_filters_frustum.cpp

```diff
@@ -86,9 +90,16 @@ Pair_Set Frustum_Filter::getFrustumIntersectionPairs() const
 #endif
   for (int i = 0; i < (int)viewIds.size(); ++i)
   {
+    // Prepare vector of intersecting objects (within loop to keep it
+    // thread-safe)
+    std::vector<HalfPlaneObject> objects = bounding_volume;
+    objects.insert(objects.end(),
```

Comment:
In the inner loop (`j`), all the elements of `objects` are constant except the last one that needs to be copied at each iteration. So the idea is to prepare `objects` with both the user-defined bounding volume and the `i`-th frustum (constant part), and reserve the last element with a dummy `HalfPlaneObject`. :+1: for using `back()` rather than the index.